jsonrpc/utils/src/format.ts

```diff
@@ -2,9 +2,37 @@ import { getError, getErrorByCode, isReservedErrorCode } from "./error";
 import { INTERNAL_ERROR, SERVER_ERROR } from "./constants";
 import { ErrorResponse, JsonRpcError, JsonRpcRequest, JsonRpcResult } from "./types";
 
+class IncrementalRandomGenerator {
+  private typedArray: Uint8Array | Uint16Array | Uint32Array;
+  private shift: number;
+  private base = 1;
+
+  static getBitsOf(typedArray: Uint8Array | Uint16Array | Uint32Array) {
+    if (!typedArray.length) {
+      throw new Error("Empty typed array");
+    }
+    return (typedArray.byteLength / typedArray.length) * 8;
+  }
+
+  constructor(bits: 8 | 16 | 32) {
+    this.typedArray =
+      bits === 8 ? new Uint8Array(1) : bits === 16 ? new Uint16Array(1) : new Uint32Array(1);
+    this.shift = IncrementalRandomGenerator.getBitsOf(this.typedArray);
+  }
+
+  getRandomValue() {
+    return (this.base++ << this.shift) + crypto.getRandomValues(this.typedArray)[0];
```

Comment:
Might need to reset `this.base` if it gets larger than `2 ** 15 - 1`

Reply:
Actually, I don't like that it has this limitation.
Let's use multiplication instead:
```suggestion
    return (this.base++ * (2 ** this.shift)) + crypto.getRandomValues(this.typedArray)[0];
```

I [measured](https://jsbm.dev/eA6lVDfZYm0bX) and it doesn't seem to have a significant performance penalty